.github/workflows/pypi_publish.yml

```diff
@@ -12,26 +13,33 @@ on:
 
 jobs:
   build-sdist:
-    name: Build package
+    name: Build sdist
     runs-on: ubuntu-latest
     steps:
       - name: 💻 Checkout the repo
         uses: actions/checkout@v4
 
       - name: 🐍 Set up Python
-        uses: actions/setup-python@v4
+        uses: conda-incubator/setup-miniconda@v3
         with:
-          python-version: '3.11'
+          miniconda-version: "latest"
+          activate-environment: build-vhdf
+          environment-file: environment.yml
+          conda-remove-defaults: true
 
-      - name: Install native dependencies
+      - name: Manually sest env variables to allow conda deps to be discoverable
+        shell: bash -l {0}
         run: |
-          set -xe
-          sudo apt update -y
-          sudo apt install -y hdf5-tools libhdf5-dev mpi-default-dev
+          CONDA_PREFIX=$(conda info --base)/envs/build-vhdf
+          echo "CONDA_PREFIX=$CONDA_PREFIX" >> $GITHUB_ENV
+          echo "$CONDA_PREFIX/bin" >> $GITHUB_PATH
+          echo "CPATH=$CONDA_PREFIX/include:$CPATH" >> $GITHUB_ENV
+          echo "LIBRARY_PATH=$CONDA_PREFIX/lib:$LIBRARY_PATH" >> $GITHUB_ENV
+          echo "LD_LIBRARY_PATH=$CONDA_PREFIX/lib:$LD_LIBRARY_PATH" >> $GITHUB_ENV
```

Comment:
This is very surprising. `setup-miniconda` _should_ take care of all this.

Reply:
I tried a couple different ways of not setting this, or setting only some of these, but this is the only way I found it works. To be honest I'd be surprised if `setup-miniconda` is setting `CPATH`, but I've spent enough time on this debugging environment variables and github context at this point that I think this is fine. The complexity of the build system is such that it's become much harder to pin down exactly what is strictly necessary and what isn't here.